extensions/azcli/src/az.ts

```diff
@@ -301,6 +303,63 @@ export class AzTool implements azExt.IAzApi {
 		};
 	}
 
+	/**
+	 * Executes az sql mi-arc show and returns a normalized object, SqlMiShowResult, regardless of the indirect or direct mode raw output shape.
+	 * @param args The args to pass to az
+	 * @param additionalEnvVars Additional environment variables to set for this execution
+	 */
+	public async executeSqlMiShow(args: string[], additionalEnvVars?: azExt.AdditionalEnvVars): Promise<azExt.AzOutput<azExt.SqlMiShowResult>> {
+		try {
+			const result = await executeAzCommand(`"${this._path}"`, args.concat(['--output', 'json']), additionalEnvVars);
+
+			let stdout = <unknown>result.stdout;
+			let stderr = <unknown>result.stderr;
+
+			try {
+				// Automatically try parsing the JSON. This is expected to fail for some az commands such as resource delete.
+				stdout = JSON.parse(result.stdout);
+			} catch (err) {
+				// If the output was not pure JSON, catch the error and log it here.
+				Logger.log(loc.azOutputParseErrorCaught(args.concat(['--output', 'json']).toString()));
```

Comment:
Do you want to continue on at this point? It'll likely fall into the else and then try to access stdout.metadata.name which will likely fail

Reply:
Good point, I'll prevent the func from continuing here